File: collect_app/src/main/java/org/odk/collect/android/preferences/utilities/PreferencesUtils.java
package org.odk.collect.android.preferences.utilities;

import androidx.preference.CheckBoxPreference;
import androidx.preference.Preference;

public final class PreferencesUtils {

    private PreferencesUtils() {

    }

    public static void displayDisabled(CheckBoxPreference preference, boolean displayValue) {
        preference.setPersistent(false);
        preference.setEnabled(false);
        preference.setChecked(displayValue);
        preference.setPersistent(true);
    }

    public static void displayDisabled(Preference preference, String displayValue) {
        preference.setEnabled(false);
        preference.setSummaryProvider(pref -> displayValue);
    }
}

File: collect_app/src/main/java/org/odk/collect/utilities/PreferencesUtils.kt
package org.odk.collect.utilities

import androidx.test.core.app.ApplicationProvider
import org.odk.collect.android.application.Collect
import org.odk.collect.android.injection.DaggerUtils
import org.odk.collect.android.preferences.PreferencesDataSource
import org.odk.collect.android.preferences.PreferencesRepository

object PreferencesUtils {
    @JvmStatic
    fun getPreferencesRepository(): PreferencesRepository {
        return DaggerUtils.getComponent(ApplicationProvider.getApplicationContext<Collect>()).preferencesRepository()
    }

    @JvmStatic
    fun getGeneralPreferences(): PreferencesDataSource {
        return getPreferencesRepository().getGeneralPreferences()
    }

    @JvmStatic
    fun getAdminPreferences(): PreferencesDataSource {
        return getPreferencesRepository().getAdminPreferences()
    }

    @JvmStatic
    fun getMetaPreferences(): PreferencesDataSource {
        return getPreferencesRepository().getMetaPreferences()
    }

    @JvmStatic
    fun getTestPreferences(name: String?): PreferencesDataSource {
        return getPreferencesRepository().getTestPreferences(name!!)
    }
}
